spec/electric_slide/call_queue_spec.rb

```diff
@@ -336,4 +332,90 @@
       end
     end
   end
+
+  describe '#update' do
+    let(:queue) {
+      ElectricSlide::CallQueue.new(
+        agent_strategy: ElectricSlide::AgentStrategy::LongestIdle,
+        connection_type: :call,
+        agent_return_method: :auto
+      )
+    }
+
+    it 'updates all the given attributes' do
+      strategy = double('Strategy').as_null_object
+      queue.update(
+        agent_strategy: strategy,
+        connection_type: :bridge,
+        agent_return_method: :manual
+      )
+      expect(queue.agent_strategy).to eq(strategy)
+      expect(queue.connection_type).to eq(:bridge)
+      expect(queue.agent_return_method).to eq(:manual)
+    end
+
+    context 'when given an unrecognized attribute' do
+      it 'does not raise an error' do
+        expect{
+          queue.update(foo: :bar)
+        }.to_not raise_exception
+      end
+    end
+
+    context 'when given `nil`' do
+      it 'does not raise an error' do
+        expect{
+          queue.update(foo: :bar)
+        }.to_not raise_exception
+      end
+    end
+  end
+
+  describe '#agent_strategy=' do
+    let(:new_strategy_class) { double('New Strategy Class').as_null_object }
+    let(:new_strategy_instance) { double('New Strategy Instance') }
+    let(:agent) { double(ElectricSlide::Agent, id: '100', presence: :available).as_null_object }
+
+    before do
+      allow(new_strategy_class).to receive(:new).and_return(new_strategy_instance)
```

Comment:
I make a (rather contrived) test for checking if the the newly-set strategy is used [here](https://github.com/neildecapia/electric_slide/blob/feature/change-queue/spec/electric_slide/call_queue_spec.rb#L391-L392). So can I keep this as long as the strategy is a proper class? Or is there a better way to test that case?